lib/engine/game/g_18_india/game.rb

```diff
@@ -790,49 +988,51 @@ def commodity_bonus(route, _stops)
           revenue = 0
           commodity_sources.each do |source|
             bonus = 0
+            @round.commodities_used = []
             case source
```

Comment:
There's a lot of repetitive code in this section, and hardcoded values. Could this be first bundling together all the information in one place:
```
COMMODITY_BONUSES = {
  'OIL' => { value: 30, commodity: 'OIL', locations: ['MUMBAI'] },
  'OPIUM' => { value: 100, commodity: 'OPIUM', locations: %w[LAHORE HALDIA] },
  'ORE1' => { value: 50, commodity: 'ORE', locations: %w[KARACHI CHENNAI] },
  # ...
}.freeze
```
And then, in `commodity_bonus`:
```
commodity_sources.each do |source|
  bonus = COMMODITY_BONUSES[source]
  if visited_names.intersect?(bonus[locations])
    revenue += bonus[value]
    @round.commoditites_used << bonus[commodity]
  end
end

Reply:
I think I understand your suggestion, I'll see if I can refactor to that approach.

Reply:
I have refactored to this method.  It is cleaner and seems to work well.